framework/src/constraints/EqualValueBoundaryConstraint.C

```diff
@@ -170,12 +170,12 @@ EqualValueBoundaryConstraint::updateConstrainedNodes()
     _mesh.getMesh().comm().allgather_packed_range(&_mesh.getMesh(),
                                                   nodes_to_ghost.begin(),
                                                   nodes_to_ghost.end(),
-                                                  mesh_inserter_iterator<Node>(_mesh.getMesh()));
+                                                  null_output_iterator<Node>());
```

Comment:
so there is a `allgather_packed_range` specialization that simply does not use this last iterator argument?

Reply:
Is there?  I admit this wouldn't be the first time I'd written an API and forgotten about it later, but everything I see in communicator.h and parallel_implementation.h requires this argument; the approx_buffer_size argument is the only optional one.

Reply:
No, this was a question. What is the point of the `null_output_iterator<Node>()` here?

Reply:
To satisfy the API we are calling. https://github.com/libMesh/libmesh/pull/3276 is the libMesh PR that necessitated this change

Reply:
Sorry I misunderstood.  The key here is what we're packing and unpacking.  With most objects the API goal was to have as much flexibility as possible - you could send a set of objects and receive into a vector or vice-versa, for example, and since we had to serialize variable-size data into buffers anyway there'd be no extra overhead.

But with Elem and Node specifically, Alex found this was actually less flexible - we were already relying on unpacking those directly into a mesh, we already didn't support doing anything else, so the iterator requirement was redundant in some cases and it was interfering with our use of more advanced APIs in other cases.  So we got rid of it - the unpack itself does the insertion and we no longer need mesh_inserter_iterator for it.

Reply:
Ok, cool, I see that now in the PR Alex linked to. The specializations are `Packing<Node *>` and `Packing<Elem *>`, not a specialization of `allgather_pached_range`.